### PR TITLE
FEAT-#1958: pin ray to 0.8.7

### LINKED ISSUE #1958: Upgrade to Ray 0.8.7
I am experiencing an issue with Ray which is fixed in `0.8.7`; however, Modin has pinned the Ray version to `0.8.6` [here](https://github.com/modin-project/modin/blob/master/modin/__init__.py#L56). Can we please update the dependency?

### PR DESCRIPTION
**Pull request by binarycrayon**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1958
- [x] tests added and passing

### CODE REVIEW

File: modin/engines/ray/utils.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def initialize_ray(
             # We only start ray in a cluster setting for the head node.
             ray.init(
                 address=redis_address or "auto",
-                include_webui=False,
+                include_dashboard=False,
```
Comment (yxue-kabam) [inline modin/engines/ray/utils.py:106]: Ray community changed `include_webui` to `include_dashboard`. see this pr https://github.com/ray-project/ray/pull/8999

Comment (binarycrayon) [inline modin/engines/ray/utils.py:106]: change from `include_webui` to `include_dashboard` due to this pr in ray community  https://github.com/ray-project/ray/pull/8999

**Comment by YarShev** [comment]
@binarycrayon , in order to run testing you need to rebase on the latest master. In addition, please update your local `black` to 20.8b1 version and run black modin.

**Comment by binarycrayon** [comment]
@YarShev got black check working, also done rebase

Screen Shot 2020-08-28 at 9 50 15 AM

not sure about failed lint check

**Comment by binarycrayon** [comment]
let me fix the commit msgs

**Comment by binarycrayon** [comment]
OK hopefully this will make the green light

**Comment by binarycrayon** [comment]
OK test_api target is failed that is expected, since it fails the check for ray version (0.8.7)

**Comment by austinmw** [comment]
Would love to get this working, can't downgrade my ray version :)

**Comment by YarShev** [comment]
@binarycrayon , thanks for updating PR! However, your commit message and name of the PR should be the following:
`FEAT-#1958: Pin ray to 0.8.7`, because your PR resolves #1958, not #8999. Also, would like to see green CI. Please, update Ray version everywhere where it is required. Also, please consider the option where we can use relaxed pin as @devin-petersohn mentioned in the #1958 .

File: modin/__init__.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def get_execution_engine():
             except ImportError:
                 pass
             else:
-                if version.parse(ray.__version__) != version.parse("0.8.6"):
+                if version.parse(ray.__version__) != version.parse("0.8.7"):
```
Comment (binarycrayon) [inline modin/__init__.py:56]: @devin-petersohn you suggested relaxing ray version check in your comment:
https://github.com/modin-project/modin/issues/1958#issuecomment-680267928

How relaxed should the check be? Should we pin it to `0.8.x` or simply check the minimal version is equal and above pip version?

**Comment by binarycrayon** [comment]
@YarShev fixed the commit message titles to ref the correct issue; also I am not sure how relaxed the ray check should be, i.e.g 0.8.x or straight `>=0.8.7`. Can I address that in a separate pr?

**Comment by YarShev** [comment]
@binarycrayon , yes, I see, thank you! Because it seems you forgot to change Ray version in [this file](https://github.com/modin-project/modin/blob/master/.github/workflows/push.yml), we can wait for @devin-petersohn and his suggestions regarding the pin.

File: modin/__init__.py (hunk 2 above)
Comment (simon-mo) [reply modin/__init__.py:56]: Hi @binarycrayon thanks for this PR. Ray is going to release 1.0 in the next version. I would suggest pinging it to 0.8.7 for now and Modin can start testing against 1.0 release candidate.

**Pull request merged**